test-data/unit/fine-grained.test

```diff
@@ -5800,3 +5800,15 @@ class M(type):
 [out]
 ==
 a.py:2: error: Argument 1 to "f" of "M" has incompatible type "int"; expected "str"
+
+[case testFString]
+from a import g
+f'{g(1)}'
```

Comment:
I would add a test where f-string is at a different scope (class or function) and/or where `g` turns into a class, if you have time.